test/pthread/test_pthread_proxying_refcount.c

```diff
@@ -11,25 +11,18 @@
 #define SANITIZER
 #endif
 
-// The first two queues will be zombies and the next two will be created just to
-// cull the zombies.
-em_proxying_queue* queues[4];
+// Proxying queues accessed from the worker thread.
+em_proxying_queue* queues[2];
 
 #ifndef SANITIZER
 
 // If we are not using sanitizers (which need to use their own allocators),
 // override free so we can track when queues are actually freed.
 
-int queues_freed[4] = {};
+_Atomic int frees = 0;
 
 void __attribute__((noinline)) free(void* ptr) {
-  for (int i = 0; i < 4; i++) {
-    if (ptr && queues[i] == ptr) {
-      queues_freed[i] = 1;
-      queues[i] = NULL;
-      break;
-    }
-  }
+  frees++;
```

Comment:
Does hit not risk counting other misc allocations?  (I guess if it ever does this test will just start failing or flaking)..

Reply:
Yeah, right now this doesn't seem to be a problem, but I could imagine that this test would start flaking in the future. At some point we might want to remove it entirely in favor of a test that builds and tests the internal `em_task_queue` implementation directly.